src/main/java/com/theelixrlabs/healthcare/service/PatientService.java

```diff
@@ -114,4 +133,25 @@ public PatientDTO getPatientById(UUID patientId) throws CustomException {
                 .patientAadhaarNumber(patientModel.getPatientAadhaarNumber())
                 .build();
     }
+
+    /**
+     * Deletes a patient by their ID, if conditions are met.
+     *
+     * @param patientId The ID of the patient to delete.
+     * @return A success message upon successful deletion.
+     * @throws CustomException If the patient is not found or is currently assigned to a doctor.
+     */
+    public String deletePatientById(String patientId) throws CustomException {
+        UUID validPatientId = validateUUID(patientId);
+        Optional<PatientModel> patientModelOptional = patientRepository.findById(validPatientId);
```

Comment:
getPatientById() can be used to perform same operation  which is present in line 145 to 148 !

Reply:
resolved it.